src/aleph/storage.py

```diff
@@ -22,113 +24,207 @@
 LOGGER = logging.getLogger("STORAGE")
 
 
-async def json_async_loads(s: str):
+class ContentSource(str, Enum):
+    DB = "DB"
+    P2P = "P2P"
+    IPFS = "IPFS"
+    INLINE = "inline"
+
+
+@dataclass
+class StoredContent:
+    hash: str
+    source: Optional[ContentSource]
+
+
+@dataclass
+class RawContent(StoredContent):
+    value: bytes
+
+    def __len__(self):
+        return len(self.value)
+
+
+@dataclass
+class JsonContent(StoredContent):
+    value: Any
```

Comment:
It could be worth defining a type for JSON compatible types instead of using Any everywhere: `Union[Dict, List, str, int, float, bool, None]`

Reply:
I tried this just now, it makes mypy angry about a lot of lines so I think I'll make the change in a different PR.

Reply:
OK